sage-impl/verify.sage

```diff
@@ -69,12 +67,28 @@ def canonical(E):
     A = min(As, key = lambda el: (int(el[0]), int(el[1])))
     return E.isomorphism_to(curve(A))
 
-def unpack_chain(A, chain):
+def unpack_chain(A, chain, deg):
     E = curve(A)
+
+    # find l-torsion basis
+    l, = deg.prime_factors()
```

Comment:
This drops all but the first prime factor, right?

Reply:
Nope, the unpacking syntax fails if the iterable on the right-hand side contains the wrong number of elements. So, this is shorthand for "make sure there is exactly one prime factor and call that `l`".

Reply:
cool that works